src/main/java/org/dependencytrack/parser/osv/GoogleOSVAdvisoryParser.java

```diff
@@ -0,0 +1,221 @@
+        }
+        return advisory;
+    }
+
+    private List<OSVVulnerability> parseVulnerabilities(JSONObject object) {
+
+        List<OSVVulnerability> osvVulnerabilityList = new ArrayList<>();
+        final JSONArray vulnerabilities = object.optJSONArray("affected");
+        if (vulnerabilities != null) {
+            for(int i=0; i<vulnerabilities.length(); i++) {
+
+                osvVulnerabilityList.addAll(parseVulnerabilityRange(vulnerabilities.getJSONObject(i)));
+            }
+        }
+        return osvVulnerabilityList;
+    }
+
+    public List<OSVVulnerability> parseVulnerabilityRange(JSONObject vulnerability) {
+
+        List<OSVVulnerability> osvVulnerabilityList = new ArrayList<>();
+        final JSONArray ranges = vulnerability.optJSONArray("ranges");
+        final JSONArray versions = vulnerability.optJSONArray("versions");
+        if (ranges != null) {
+            for (int j=0; j<ranges.length(); j++) {
+                final JSONObject range = ranges.getJSONObject(j);
+                String rangeType = range.optString("type", null);
+                if(rangeType != null && !rangeType.equalsIgnoreCase("GIT")) {
+                    osvVulnerabilityList.addAll(parseVersionRanges(vulnerability, range));
+                }
+            }
+        }
+        // if ranges are not available or only commit hash range is available, look for versions
+        if (osvVulnerabilityList.size() == 0 && versions != null && versions.length() > 0) {
+            for (int j=0; j<versions.length(); j++) {
+                OSVVulnerability vuln = createOSVVulnerability(vulnerability);
+                vuln.setVersion(versions.getString(j));
+                osvVulnerabilityList.add(vuln);
+            }
+        }
+        // if no parsable range or version is avilable, add vulnerability without version
+        else if (osvVulnerabilityList.size() == 0) {
+            osvVulnerabilityList.add(createOSVVulnerability(vulnerability));
+        }
+        return osvVulnerabilityList;
+    }
+
+    private List<OSVVulnerability> parseVersionRanges(JSONObject vulnerability, JSONObject range) {
+
+        final List<OSVVulnerability> osvVulnerabilityList = new ArrayList<>();
+        final JSONArray rangeEvents = range.optJSONArray("events");
+        if(rangeEvents != null) {
+            int k = 0;
+            while (k < rangeEvents.length()) {
+
+                OSVVulnerability osvVulnerability = createOSVVulnerability(vulnerability);
+                JSONObject event = rangeEvents.getJSONObject(k);
+                String lower = event.optString("introduced", null);
+                if(lower != null) {
+                    osvVulnerability.setLowerVersionRange(lower);
+                    k += 1;
+                }
+                if(k < rangeEvents.length()) {
+                    event = rangeEvents.getJSONObject(k);
+                    String upper = event.optString("fixed", null);
```

Comment:
We'll need to handle `limit` and `last_affected` events as well:

<img width="889" alt="image" src="https://user-images.githubusercontent.com/5693141/176761981-903b363c-ed0a-4a25-b37b-7a2e23f9d58d.png">